parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/state/WorkManager.java

```diff
@@ -58,19 +60,28 @@
 
     @Getter
     private int numberRecordsOutForProcessing = 0;
-
+    private PCModule<K,V> module;
     /**
      * Useful for testing
      */
     @Getter(PUBLIC)
     private final List<Consumer<WorkContainer<K, V>>> successfulWorkListeners = new ArrayList<>();
 
+    private Gauge inflightRecordsMeter;
+    private Gauge waitingRecordsMeter;
+
     public WorkManager(PCModule<K, V> module,
                        DynamicLoadFactor dynamicExtraLoadFactor) {
+        this.module = module;
         this.options = module.options();
         this.dynamicLoadFactor = dynamicExtraLoadFactor;
         this.sm = new ShardManager<>(module, this);
         this.pm = new PartitionStateManager<>(module, sm);
+
+        inflightRecordsMeter = Gauge.builder("pc.records.inflight", this::getNumberRecordsOutForProcessing)
+                .register(module.meterRegistry());
+        waitingRecordsMeter = Gauge.builder("pc.records.waiting", this::getNumberOfWorkQueuedInShardsAwaitingSelection)
```

Comment:
should depend on getNumberOfWorkQueuedInShardsAwaitingSelection performance fixes from GPT branch?